src/bootstrap/bootstrap.py

```diff
@@ -110,29 +110,42 @@ def download(path, url, probably_big, verbose, help_on_error=None):
 
 
 def _download(path, url, probably_big, verbose, exception, help_on_error=None):
+    # Try to use curl (potentially available on win32
+    #    https://devblogs.microsoft.com/commandline/tar-and-curl-come-to-windows/)
+    # If an error occurs:
+    #  - If we are on win32 fallback to powershell
+    #  - Otherwise raise the error if appropriate
     if probably_big or verbose:
         print("downloading {}".format(url))
-    # see https://serverfault.com/questions/301128/how-to-download
-    if sys.platform == 'win32':
-        run(["PowerShell.exe", "/nologo", "-Command",
-             "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;",
-             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')".format(url, path)],
-            verbose=verbose,
-            exception=exception)
-    else:
+
+    platform_is_win32 = sys.platform == 'win32'
+    try:
         if probably_big or verbose:
             option = "-#"
         else:
             option = "-s"
-        require(["curl", "--version"])
+        # If curl is not present on Win32, we shoud not sys.exit
+        #   but raise `CalledProcessError` or `OSError` instead
+        require(["curl", "--version"], exception=platform_is_win32)
         run(["curl", option,
              "-L", # Follow redirect.
              "-y", "30", "-Y", "10",    # timeout if speed is < 10 bytes/sec for > 30 seconds
              "--connect-timeout", "30",  # timeout if cannot connect within 30 seconds
              "--retry", "3", "-Sf", "-o", path, url],
             verbose=verbose,
-            exception=exception,
+            exception=True, # Will raise RuntimeError on failure
             help_on_error=help_on_error)
+    except (subprocess.CalledProcessError, OSError, RuntimeError):
+        # see http://serverfault.com/questions/301128/how-to-download
+        if platform_is_win32:
+            run(["PowerShell.exe", "/nologo", "-Command",
+                 "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;",
+                 "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')".format(url, path)],
+                verbose=verbose,
+                exception=exception)
+        # Check if the RuntimeError raised by run(curl) should be silenced
+        elif verbose or exception:
+            raise RuntimeError
```

Comment:
It seems like we should be trying to re-raise the original exception -- this looks like it'll probably create a new exception instead?

Reply:
TLDR: Good catch! I will remove the `RuntimeError`.

Long note:
In python3  the traceback is stored in the exception:
```
$cat test.py
def foo():
	raise RuntimeError("foo error")
try:
	foo()
except RuntimeError:
	raise RuntimeError
$python3 test.py
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    foo()
  File "test.py", line 2, in foo
    raise RuntimeError("foo error")
RuntimeError: foo error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 7, in <module>
    raise RuntimeError
RuntimeError
```

But the traceback is ugly. With just `raise` everything is nice and tidy
 ```
$cat test.py
def foo():
	raise RuntimeError("foo error")
try:
	foo()
except RuntimeError:
	raise
$python3 test.py
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    foo()
  File "test.py", line 2, in foo
    raise RuntimeError("foo error")
RuntimeError: foo error
(base) razoa@hagarth:~/ru
```

Thanks!